v1/related.yaml

```diff
@@ -0,0 +1,133 @@
+swagger: '2.0'
+info:
+  version: '1.0.0-beta'
+  title: MediaWiki Related Pages API
+  description: Related pages API
+  termsofservice: https://github.com/wikimedia/restbase#restbase
+  contact:
+    name: Services
+    email: services@lists.wikimedia.org
+    url: https://www.mediawiki.org/wiki/Services
+  license:
+    name: Apache licence, v2
+    url: https://www.apache.org/licenses/LICENSE-2.0
+paths:
+  /related/{title}:
+    get:
+      tags:
+        - Page content
+      summary: Get pages related to the given title
+      description: |
+        Returns summaries for 5 pages related to the given page. Summaries include
+        page title, namespace and id along with short text description of the page
+        and a thumbnail.
+
+        Stability: [experimental](https://www.mediawiki.org/wiki/API_versioning#Experimental)
+      produces:
+        - application/json
+      parameters:
+        - name: title
+          in: path
+          description: The page title.
+          type: string
+          required: true
+      responses:
+        '200':
+          description: The related pages
+          schema:
+            $ref: '#/definitions/related'
+        '404':
+          description: Unknown page title
+          schema:
+            $ref: '#/definitions/problem'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+
+      x-request-handler:
+        # Storage miss. Call the Action API to get the textextract.
+        - api_request:
+            request:
+              method: 'post'
+              uri: /{domain}/sys/action/query
+              body:
+                format: 'json'
+                prop: 'pageterms|pageimages|pageprops'
+                ppprop: 'mainpage|disambiguation'
+                wbptterms: 'description'
+                generator: 'search'
+                gsrsearch: 'morelike:{request.params.title}'
+                gsrnamespace: 0
+                gsrwhat: 'text'
+                gsrinfo: ''
+                gsrprop: 'redirecttitle'
+                gsrlimit: 5
+                piprop: 'thumbnail'
+                pithumbsize: 640
```

Comment:
Correction. It should use 320px but for some reason it uses 640px, as Michael mentioned earlier. I'm not sure why because the thumbnails in read more seem to have the same size as for search, history, saved pages.